### ISSUE TITLE
Bug - Library name missing - Plex media not found using payload: '{"title":"Toy Story"}'

### ISSUE THREAD

**Issue by bremor** [JurajNyiri/PlexMeetsHomeAssistant#26]
**Describe the bug**
When casting to a chromecast, it appears that the play_media service call fails because the library_name is not specified. It seems to fall back to casting as a file, which means i lose all cover art/metadata information.

**To Reproduce**
Cast a movie to a chromecast.

**Checks**
Please check below if true:

[x] I am running Home Assistant on HTTP protocol
[ ] I am running BOTH Home Assistant AND Plex on HTTPS protocol
[x] My Plex server is accessible through protocol, port and ip/hostname I have specified for the card.

I believe the issue is related to this peice of code, data.librarySectionTitle does not seem to be valid and is thus omitted within JSON.stringify.
```
                                case 'movie':
                                    await this.playViaCastPlex(entity.value, 'movie', `plex://${JSON.stringify({
                                        // eslint-disable-next-line @typescript-eslint/camelcase
                                        library_name: data.librarySectionTitle,
                                        title: data.title
                                    })}`);
                                    break;
```

**Comment by JurajNyiri** [comment]
Hello, please put in console.log(data.librarySectionTitle) and console.log(data) before break statement. Make sure to delete any .gz files next to .js after doing so and clear browser cache and refresh page 2 times.

Please post results so I can figure out what exactly is wrong.

**Comment by bremor** [comment]
Thanks for your response :)

librarySectiontitle is undefined. data is shown, there doesn't appear to be any reference to the library name.

`{
    "ratingKey": "150",
    "key": "/library/metadata/150",
    "guid": "plex://movie/5d776828880197001ec90d44",
    "studio": "Pixar",
    "type": "movie",
    "title": "Toy Story",
    "contentRating": "au/G",
    "summary": "Led by Woody, Andy's toys live happily in his room until Andy's birthday brings Buzz Lightyear onto the scene. Afraid of losing his place in Andy's heart, Woody plots against Buzz. But when circumstances separate Buzz and Woody from their owner, the duo eventually learns to put aside their differences.",
    "rating": 10,
    "audienceRating": 9.2,
    "year": 1995,
    "tagline": "\"Oooh...3-D\" (USA 2009 re-release)",
    "thumb": "/library/metadata/150/thumb/1627880444",
    "art": "/library/metadata/150/art/1627880444",
    "duration": 4863567,
    "originallyAvailableAt": "1995-11-22",
    "addedAt": 1627880441,
    "updatedAt": 1627880444,
    "audienceRatingImage": "rottentomatoes://image.rating.upright",
    "hasPremiumPrimaryExtra": "1",
    "ratingImage": "rottentomatoes://image.rating.ripe",
    "Media": [
        {
            "id": 394,
            "duration": 4863567,
            "bitrate": 1937,
            "width": 1920,
            "height": 1080,
            "aspectRatio": 1.78,
            "audioChannels": 2,
            "audioCodec": "aac",
            "videoCodec": "h264",
            "videoResolution": "1080",
            "container": "mp4",
            "videoFrameRate": "24p",
            "optimizedForStreaming": 1,
            "audioProfile": "lc",
            "has64bitOffsets": false,
            "videoProfile": "high",
            "Part": [
                {
                    "id": 394,
                    "key": "/library/parts/394/1627352596/file.mp4",
                    "duration": 4863567,
                    "file": "/kids-movies/Toy Story (1995).mp4",
                    "size": 1179938215,
                    "audioProfile": "lc",
                    "container": "mp4",
                    "has64bitOffsets": false,
                    "optimizedForStreaming": true,
                    "videoProfile": "high"
                }
            ]
        }
    ],
    "Genre": [
        {
            "tag": "Animation"
        },
        {
            "tag": "Comedy"
        }
    ],
    "Director": [
        {
            "tag": "John Lasseter"
        }
    ],
    "Writer": [
        {
            "tag": "Andrew Stanton"
        },
        {
            "tag": "John Lasseter"
        }
    ],
    "Country": [
        {
            "tag": "United States of America"
        }
    ],
    "Role": [
        {
            "tag": "Tom Hanks"
        },
        {
            "tag": "Tim Allen"
        },
        {
            "tag": "Don Rickles"
        }
    ]
}`

After some experimentation, I was able to fix this on my system by passing `this.config.libraryName` into `new PlayController` and then referencing that value when making the play_media service_call. I'm happy to make a pull request to fix, but you might have a more clever or 'correct' way to fix this.

**Comment by JurajNyiri** [comment]
It was made the way you are suggesting when originally implemented ( https://github.com/JurajNyiri/PlexMeetsHomeAssistant/commit/25d9dadb9a76c1041272e3622051545fa57118ff ) but there is an issue with it that it doesn't work in special libraries like recently added, deck etc. If you pass that Home Assistant doesn't know that library. 

Does some kind of reference appear for you when you use special library like deck or continue watching?

I will have to investigate this further and figure out a correct solution that would work for both cases.

**Comment by bremor** [comment]
Here is the same movie played two different ways... the first one using the special library "Watch Next" and the other using the actual library. When using a normal library, librarySectionTitle is missing. You could check if librarySectionTitle is missing, and if it's missing you could use the libraryName that is passed into the PlayController object.
`{
    "ratingKey": "149",
    "key": "/library/metadata/149",
    "guid": "plex://movie/5d7770af51dd69001fe59db2",
    "studio": "Magic Light Pictures",
    "type": "movie",
    "title": "Zog",
    "librarySectionTitle": "Kid's Movies",
    "librarySectionID": 1,
    "librarySectionKey": "/library/sections/1",
    "contentRating": "G",
    "summary": "Zog is a keen young dragon who wants to learn to fly, roar, and breathe fire, so he studies at the Dragon School. He is desperate to impress his teacher and win a golden star, but he is prone to accidents and he has to be helped by a kind young girl who patches up his bumps and bruises.",
    "audienceRating": 7.5,
    "viewOffset": 216000,
    "skipCount": 1,
    "lastViewedAt": 1628165641,
    "includedAt": 1628165641,
    "year": 2018,
    "tagline": "Even Dragons Have To Go To School",
    "thumb": "/library/metadata/149/thumb/1627351808",
    "art": "/library/metadata/149/art/1627351808",
    "duration": 1621080,
    "originallyAvailableAt": "2018-12-25",
    "addedAt": 1575783179,
    "updatedAt": 1627351808,
    "audienceRatingImage": "rottentomatoes://image.rating.upright",
    "Media": [
        {
            "id": 393,
            "duration": 1621080,
            "bitrate": 1210,
            "width": 1024,
            "height": 576,
            "aspectRatio": 1.78,
            "audioChannels": 2,
            "audioCodec": "aac",
            "videoCodec": "h264",
            "videoResolution": "576",
            "container": "mp4",
            "videoFrameRate": "PAL",
            "optimizedForStreaming": 0,
            "audioProfile": "lc",
            "has64bitOffsets": false,
            "videoProfile": "high",
            "Part": [
                {
                    "id": 393,
                    "key": "/library/parts/393/1575783179/file.mp4",
                    "duration": 1621080,
                    "file": "/kids-movies/Zog (2018).mp4",
                    "size": 246778296,
                    "audioProfile": "lc",
                    "container": "mp4",
                    "has64bitOffsets": false,
                    "optimizedForStreaming": false,
                    "videoProfile": "high"
                }
            ]
        }
    ],
    "Genre": [
        {
            "tag": "Animation"
        },
        {
            "tag": "Comedy"
        }
    ],
    "Director": [
        {
            "tag": "Max Lang"
        },
        {
            "tag": "Daniel Snaddon"
        }
    ],
    "Writer": [
        {
            "tag": "Max Lang"
        },
        {
            "tag": "Suzanne Lang"
        }
    ],
    "Country": [
        {
            "tag": "United Kingdom"
        }
    ],
    "Role": [
        {
            "tag": "Lenny Henry"
        },
        {
            "tag": "Hugh Skinner"
        },
        {
            "tag": "Tracey Ullman"
        }
    ]
}`


`{
    "ratingKey": "149",
    "key": "/library/metadata/149",
    "guid": "plex://movie/5d7770af51dd69001fe59db2",
    "studio": "Magic Light Pictures",
    "type": "movie",
    "title": "Zog",
    "contentRating": "G",
    "summary": "Zog is a keen young dragon who wants to learn to fly, roar, and breathe fire, so he studies at the Dragon School. He is desperate to impress his teacher and win a golden star, but he is prone to accidents and he has to be helped by a kind young girl who patches up his bumps and bruises.",
    "audienceRating": 7.5,
    "viewOffset": 216000,
    "skipCount": 1,
    "lastViewedAt": 1628165641,
    "year": 2018,
    "tagline": "Even Dragons Have To Go To School",
    "thumb": "/library/metadata/149/thumb/1627351808",
    "art": "/library/metadata/149/art/1627351808",
    "duration": 1621080,
    "originallyAvailableAt": "2018-12-25",
    "addedAt": 1575783179,
    "updatedAt": 1627351808,
    "audienceRatingImage": "rottentomatoes://image.rating.upright",
    "Media": [
        {
            "id": 393,
            "duration": 1621080,
            "bitrate": 1210,
            "width": 1024,
            "height": 576,
            "aspectRatio": 1.78,
            "audioChannels": 2,
            "audioCodec": "aac",
            "videoCodec": "h264",
            "videoResolution": "576",
            "container": "mp4",
            "videoFrameRate": "PAL",
            "optimizedForStreaming": 0,
            "audioProfile": "lc",
            "has64bitOffsets": false,
            "videoProfile": "high",
            "Part": [
                {
                    "id": 393,
                    "key": "/library/parts/393/1575783179/file.mp4",
                    "duration": 1621080,
                    "file": "/kids-movies/Zog (2018).mp4",
                    "size": 246778296,
                    "audioProfile": "lc",
                    "container": "mp4",
                    "has64bitOffsets": false,
                    "optimizedForStreaming": false,
                    "videoProfile": "high"
                }
            ]
        }
    ],
    "Genre": [
        {
            "tag": "Animation"
        },
        {
            "tag": "Comedy"
        }
    ],
    "Director": [
        {
            "tag": "Max Lang"
        },
        {
            "tag": "Daniel Snaddon"
        }
    ],
    "Writer": [
        {
            "tag": "Max Lang"
        },
        {
            "tag": "Suzanne Lang"
        }
    ],
    "Country": [
        {
            "tag": "United Kingdom"
        }
    ],
    "Role": [
        {
            "tag": "Lenny Henry"
        },
        {
            "tag": "Hugh Skinner"
        },
        {
            "tag": "Tracey Ullman"
        }
    ]
}`

**Comment by JurajNyiri** [comment]
Fixed in 2.0.12.
